app/controllers/grades_controller.rb

```diff
@@ -40,7 +40,7 @@ def to_csv
     respond_to do |format|
       format.csv do
         send_data Grade.to_csv(params[:assignment_id].to_i),
-                  filename: "assignment-grades.csv"
+                  filename: "#{@assignment.name}.csv"
```

Comment:
Just do `Assignment.find(params[:assignment_id]).name` and please test PR. Don't send PR without testing.

Reply:
Sorry for the inconvenience, I will fix this and also upload screenshots after testing.